cli/install/install.go

```diff
@@ -424,30 +462,53 @@ func installTt(binDir string, installCtx InstallCtx, distfiles string) error {
 	// The tag format in tt is vX.Y.Z, but the user can use the X.Y.Z format
 	// and this option needs to be supported.
 	if ttVersion != "master" {
-		versionFound := false
-		for _, ver := range versions {
-			if ttVersion == ver.Str || (ttVersion[0:1] != "v" && "v"+ttVersion == ver.Str) {
-				versionFound = true
-				ttVersion = ver.Str
-				break
+		if isVersion(ttVersion) {
+			for _, ver := range versions {
+				if ttVersion == ver.Str || (ttVersion[0:1] != "v" && "v"+ttVersion == ver.Str) {
+					versionFound = true
+					ttVersion = ver.Str
+					break
+				}
+			}
+		} else {
+
+			log.Infof("Searching in commits...")
+
+			if len(ttVersion) < minCommitHashLength {
+				return fmt.Errorf("the hash must contain at least 7 characters")
+			}
+
+			commits, err := getCommits(installCtx.Local, distfiles, "tt")
+
+			if err != nil {
+				return err
+			}
+
+			croppedHash := ttVersion[0:minCommitHashLength]
+
+			if slices.Contains(commits, croppedHash) {
+				hashFound = true
+				ttVersion = croppedHash
 			}
 		}
 
-		if !versionFound {
-			return fmt.Errorf("%s version of tt doesn't exist", ttVersion)
+		if !hashFound && !versionFound {
+			return fmt.Errorf("%s version or commit of tt doesn't exist", ttVersion)
 		}
 	}
 
 	// Check binary directory.
 	if binDir == "" {
 		return fmt.Errorf("binDir is not set, check %s", configure.ConfigName)
 	}
+
```

Comment:
This change is not related to the feature implementation.

Reply:
fixed